contracts/PodCastNFT.sol

```diff
@@ -5,53 +5,60 @@ import "hardhat/console.sol";
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
-import "./ConsensusAdminable.sol";
 
 import {Base64} from "./libraries/Base64.sol";
 
-contract PodCastNFT is ERC721URIStorage, Ownable, ConsensusAdminable {
+contract PodCastNFT is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
 
-    constructor(address[] memory _admin, bool givenHighestAuthority)
-        ConsensusAdminable(_admin, givenHighestAuthority)
-        ERC721("Henkaku v0.1", "henkaku")
-    {}
+    constructor() ERC721("Henkaku v0.2", "henkaku") {}
 
+    mapping(address => string[]) public roles;
     mapping(uint256 => bool) private _communityMemberShip;
 
-    function isCommunityMember(uint256 _tokenId) public view returns (bool) {
-        return _communityMemberShip[_tokenId];
+    function getRoles(address _address) public view returns (string[] memory) {
+        require(balanceOf(_address) != 0, "wallet must have membership nft");
+        return roles[_address];
     }
 
-    function isCommunityMemberByCommunityRole(string memory _roleInCommunity)
-        internal
-        pure
+    function hasRoleOf(address _address, string memory _role)
+        public
+        view
         returns (bool)
     {
-        return bytes(_roleInCommunity).length > 2;
+        require(balanceOf(_address) != 0, "wallet must have membership nft");
+        string[] memory _roles = roles[_address];
+        for (uint256 i = 0; i < _roles.length; i++) {
+            if (keccak256(bytes(_roles[i])) == keccak256(bytes(_role))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function setRoles(address _to, string[] memory _roles) public onlyOwner {
+        require(balanceOf(_to) != 0, "wallet must have membership nft");
+        roles[_to] = _roles;
     }
 
+    function addRole(address _to, string memory _role) public onlyOwner {
+        require(balanceOf(_to) != 0, "wallet must have membership nft");
+        roles[_to].push(_role);
+    }
+
+    function updateOwnNFT(string memory _imageURI, string memory name) public {}
+
     function updateNFT(
```

Comment:
You may need to do `roles[_to] = _roles;` here.

As is the case with mint, I am wondering whether it is better to write `roles[_to] = _roles; ` or read `setRoles` in such a case.

Reply:
hmm, normally we don't update roles often.
so I thought separation would be nice.
if we need to call `setRoles`, then we all the time need to specify array correctly in client side.
which is likely causing human error

Reply:
Roger, I feel that updateNFT should not accept the role change as a parameter if it is not needed.

In that case, the image, will still be changed, so I wonder if the process will be to load the role for that address.

Reply:
Owner is not the NFT holder but initially Contract deployer is the owner